nano/nano_wallet/entry.cpp

```diff
@@ -10,10 +10,15 @@
 #include <nano/qt/qt.hpp>
 
 #include <boost/make_shared.hpp>
-#include <boost/process.hpp>
 #include <boost/program_options.hpp>
 #include <boost/property_tree/json_parser.hpp>
 #include <boost/property_tree/ptree.hpp>
+#include <boost/version.hpp>
+
+#define USE_BOOST_PROCESS BOOST_VERSION != 106900
```

Comment:
Does this problem happen on anything but macOS? If it's mac only, we could add that to the constraint so that 1.69 works on Linux (which most services use) and Windows.

Reply:
This is just for the gui wallet, I don't think many people build this from source do they? (could be wrong) so I played it safe as didn't check all platforms to see whether it failed to build with boost 1.69.

Reply:
Oh ok, so daemon doesn't have the same check (IOW daemon won't compile with 1.69 on macOS?)

Reply:
daemon will compile as it doesn't use the boost process, it does things a bit differently (see my first comment). It has to do with proper shutdown of the child RPC process. When clicking the X (close) button on the gui wallet, we have no way to terminate the RPC process as the standard doesn't know about processes yet, so it leaves dangling invisible processes. We could probably set up a connection with the RPC process from the daemon and send the stop RPC command, but I just prevent the child process being opened if using boost 1.69 and say they have to start it separately. With the node daemon, it is either closed by sending `stop` RPC command or forcefully quitting, which seems to clean up the child process, although I can imagine in some circumstances this might not do it correctly.

Reply:
OK, I noticed the #include <boost/process.hpp> in the daemon while reviewing. I haven't compiled the latest yet - but the include is needed?

Reply:
it should be removed in the latest (I can't see it at least), including that header or pretty much any sub-header will indeed fail to compile on mac with 1.69

Reply:
Boost interprocess stuff works with Windows + 1.69 so I have added an additional compilation check and allow it for windows. Have raised #1879 to check for linux. I also found under some circumstances (like forcefully exiting a debug session), the child rpc process is not cleaned up on windows with the CLI daemon, so I will add similar checks there as well (after #1879) to know which platforms are supported.